corehq/apps/app_manager/static/app_manager/js/app_manager.js

```diff
@@ -240,46 +263,27 @@ hqDefine('app_manager/js/app_manager.js', function () {
      * @private
      */
     var _initNewItemForm = function () {
-        $('.new-module').on('click', function (e) {
-            e.preventDefault();
-            var dataType = $(this).data('type');
-            $('#new-module-type').val(dataType);
-            var form = $('#new-module-form');
-            if (!form.data('clicked')) {
-                form.data('clicked', 'true');
-                $('.new-module-icon').removeClass().addClass("fa fa-refresh fa-spin");
-                form.submit();
-            }
-        });
-
-        $('.new-form').on('click', function (e) {
-            e.preventDefault();
-            var dataType = $(this).data("type");
-            var moduleId = $(this).data("module");
-            var form = $("#form-to-create-new-form-for-module-" + moduleId);
-            $("input[name=form_type]", form).val(dataType);
-            if (!form.data('clicked')) {
-                form.data('clicked', 'true');
-                $('.new-form-icon-module-' + moduleId).removeClass().addClass("fa fa-refresh fa-spin");
-                form.submit();
-            }
-        });
-        if (COMMCAREHQ.toggleEnabled('APP_MANAGER_V2')) {
-            $(document).on('click', '.js-new-form', function (e) {
+        if (!COMMCAREHQ.toggleEnabled('APP_MANAGER_V2')) {
+            $('.new-module').on('click', function (e) {
                 e.preventDefault();
-                var $a = $(this),
-                    $popoverContent = $a.closest(".popover-content > *"),
-                    $form = $popoverContent.find("form"),
-                    action = $a.data("case-action"),
-                    moduleId = $popoverContent.data("module-id"),
-                    $trigger = $('.js-add-new-item[data-module-id="' + moduleId + '"]');
-                $form.attr("action", hqImport("hqwebapp/js/urllib.js").reverse("new_form", moduleId));
-                $form.find("input[name='case_action']").val(action);
-                $form.find("input[name='form_type']").val($a.data("form-type"));
-                if (!$form.data('clicked')) {
-                    $form.data('clicked', 'true');
-                    $trigger.find(".fa-plus").removeClass("fa-plus").addClass("fa fa-refresh fa-spin");
-                    $form.submit();
+                var dataType = $(this).data('type');
+                $('#new-module-type').val(dataType);
+                var form = $('#new-module-form');
+                if (!form.data('clicked')) {
+                    form.data('clicked', 'true');
+                    $('.new-module-icon').removeClass().addClass("fa fa-refresh fa-spin");
+                    form.submit();
+                }
+            });
+            $('.new-form').on('click', function (e) {
+                var dataType = $(this).data("type");
+                var moduleId = $(this).data("module");
+                var form = $("#form-to-create-new-form-for-module-" + moduleId);
```

Comment:
Does this work right? It looks like `.new-form` uses the index-based id (https://github.com/dimagi/commcare-hq/blob/581fce55f27b3677a6701969a06d7cfbac1e06aa/corehq/apps/app_manager/templates/app_manager/v1/managed_app.html#L206) but `#form-to-create-new-form-for-module-X` uses the unique id (https://github.com/dimagi/commcare-hq/blob/581fce55f27b3677a6701969a06d7cfbac1e06aa/corehq/apps/app_manager/templates/app_manager/v1/managed_app.html#L193).